app/scenes/Settings/components/PeopleTable.tsx

```diff
@@ -1,99 +1,113 @@
-import { observer } from "mobx-react";
+import compact from "lodash/compact";
 import * as React from "react";
 import { useTranslation } from "react-i18next";
 import styled from "styled-components";
 import User from "~/models/User";
 import { Avatar } from "~/components/Avatar";
 import Badge from "~/components/Badge";
 import Flex from "~/components/Flex";
-import TableFromParams from "~/components/TableFromParams";
+import {
+  type Props as TableProps,
+  SortableTable,
+} from "~/components/SortableTable";
+import { type Column as TableColumn } from "~/components/Table";
 import Time from "~/components/Time";
 import useCurrentUser from "~/hooks/useCurrentUser";
 import UserMenu from "~/menus/UserMenu";
 
-type Props = Omit<React.ComponentProps<typeof TableFromParams>, "columns"> & {
-  data: User[];
+type Props = Omit<TableProps<User>, "columns" | "rowHeight" | "gridColumns"> & {
   canManage: boolean;
 };
 
-function PeopleTable({ canManage, ...rest }: Props) {
+export function PeopleTable({ canManage, ...rest }: Props) {
   const { t } = useTranslation();
   const currentUser = useCurrentUser();
-  const columns = React.useMemo(
+
+  const gridColumns = React.useMemo(() => {
+    if (canManage) {
+      return "4fr 4fr 2fr 2fr 0.5fr"; // all columns will be displayed.
```

Comment:
This grid becomes necessary since absolutely positioned table rows are not stylable to take up the full width. Instead, the rows are rendered as grid, with the cells being the grid items.

Reply:
Hmmm yea I really don't like this defined here – the css should be derived from the columns within the Table component somehow. We can define some sort of `width` property on the column definition below…?

Reply:
Do you mean to skip the whole grid styles? It becomes tricky to accept percentage as column width since we need to ensure all the columns add up to 100% for a full-width table.
The simpler option would be to move the individual parts of `4fr 4fr 2fr 2fr 0.5fr` to its column def. Would that be ok?

Reply:
Grid style is fine, with grid you can an use mixed widths so one column could be a fixed px and another could be `fr` unit? I think it will work okay if we can move to config